### PR TITLE
feat: add login interface with token

### PR DESCRIPTION
**Pull request by aon**
# What ❔

- Add login view with token generation
- Remove login button functionality
- Add proxy feature to allow token generation
- Add RPC proxy (temporary) feature to passthrough all RPC commands.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.

### CODE REVIEW

File: packages/app/src/configs/dev.config.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
       ],
       "icon": "/images/icons/zksync-arrows.svg",
-      "l2ChainId": 270,
+      "l2ChainId": 300,
       "l2NetworkName": "Local",
       "maintenance": false,
       "name": "local",
       "published": true,
-      "rpcUrl": "http://localhost:3050",
+      "rpcBaseUrl": "http://localhost:4041/rpc",
```
Comment (calvogenerico) [inline packages/app/src/configs/dev.config.json:15]: Why this name change? The explorer is still interacting with a single rpc, right?

File: packages/proxy/src/routes/auth.ts [hunk 2]
```diff
@@ -30,21 +31,33 @@ export default function authRoutes(app: FastifyApp) {
...
+      return reply
+        .status(400)
+        .send({ message: 'Failed to verify signed message.' });
     }
   });
 
   app.get('/logout', async (req, reply) => {
     deleteAuthSession(req);
-    reply.status(204).send();
+    return reply.status(204).send();
```
Comment (calvogenerico) [inline packages/proxy/src/routes/auth.ts:45]: omg, we were doing this everywhere! Nice catch.

File: packages/proxy/src/services/rpc.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export async function getContractOwner(
   address: Address,
 ): Promise<Address | null> {
   try {
-    return await client.readContract({
+    return client.readContract({
```
Comment (calvogenerico) [inline packages/proxy/src/services/rpc.ts:16]: I feel that that this is one of the weird cases were we want the `return await`.

Inside a `try/catch` doing `return ...` or `return await ...` has different behavior.

Look at this code:

``` jsasync function throwError() {
    throw new Error("hi, this is an error")
}

async function withAwait() {
    try {
        return await throwError()
    } catch (e) {
        console.log("catched with await!")
    }
    
}

async function noAwait() {
    try {
        return throwError()
    } catch (e) {
        console.log("catched with no await!")
    }
}


withAwait().then(() => console.log("No error was thrown"))
noAwait().catch(() => console.log("An error was thrown"))
```
Executing this prints:

``` text
catched with await!
No error was thrown
An error was thrown
```

Only the awaited one actually catch the error inside the try catch. I believe that in this case we want to catch the error, so we need the await.

**Review by calvogenerico — COMMENTED** [review]
Looks amazing. The comments are minor, but in particular the one about the await I believe that we want it in.

File: packages/proxy/src/services/rpc.ts (hunk 3 above)
Comment (aon) [reply packages/proxy/src/services/rpc.ts:16]: 100%, you're right.

File: packages/private-rpc/src/schemas/address.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  try {
+    return getAddress(val);
+  } catch {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: 'Invalid address',
+    });
+    return z.NEVER;
+  }
+});
```
Comment (calvogenerico) [inline packages/private-rpc/src/schemas/address.ts:14]: :ok_hand:

**Review by calvogenerico — APPROVED** [review]

**Pull request merged**